lib/web/integrations.go

```diff
@@ -219,6 +223,79 @@ func (h *Handler) integrationsDelete(w http.ResponseWriter, r *http.Request, p h
 	return OK(), nil
 }
 
+// integrationsDelete removes an Integration based on its name
+func (h *Handler) integrationsDeleteWithCleanup(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
+	values := r.URL.Query()
+	orgName := values.Get("orgName")
+
+	integrationName := values.Get("name")
+	if integrationName == "" {
+		return nil, trace.BadParameter("an integration name is required")
+	}
+
+	integrationKind := values.Get("subKind")
+	if integrationKind == "" {
+		return nil, trace.BadParameter("the integrations subKind is required")
+	}
+
+	// Validate subKind support.
+	switch integrationKind {
+	case types.KindGithub:
+		if orgName == "" {
+			return nil, trace.BadParameter("the integrations orgName is required")
+		}
+	default:
+		return nil, trace.BadParameter("deleting integration with clean up for subKind %q not supported", integrationKind)
+	}
+
+	clt, err := sctx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := clt.DeleteIntegration(r.Context(), integrationName); err != nil {
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err)
+		}
+		// Ignore not found errors. It could have already been deleted if a user
+		// is retrying from failure to delete git servers for whatever reason.
+	}
+
+	if integrationKind == types.KindGithub {
+		// Query for all git servers that was servicing this integration.
+		queryReq := &proto.ListUnifiedResourcesRequest{
+			Kinds:               []string{types.KindGitServer},
+			SortBy:              types.SortBy{Field: "name"},
+			PredicateExpression: fmt.Sprintf(`labels["%s"] == "%s"`, types.GitHubOrgLabel, orgName),
+			// There shouldn't be more than 1k of git servers for the same integration.
+			Limit: defaults.MaxIterationLimit,
+		}
+
+		resources, _, err := apiclient.GetUnifiedResourcePage(r.Context(), clt, queryReq)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		if len(resources) == 0 {
+			return nil, trace.NotFound("Integration was deleted successfully but no Git servers were deleted. Git servers for organization %q was not found.", orgName)
+		}
```

Comment:
Should this actually be an error?

If I create an integration but decide to delete it before it imports any resources, I would expect the delete operation to succeed, not show me an error.

Reply:
the enrolling flow creates the git server right after it creates the integration in one step. it also creates the role required to be assigned to have access to the created git server.

i ran into this issue myself where, i deleted the integration right after creating (before I assigned myself the role) and the git server wasn't deleted because I had no access to it. so I thought it would be good to warn the user that it wasn't deleted for cases like that

Reply:
it's been decided to prevent users from deleting the integration if user can't access some of the git servers related to integration